### PR TITLE
Upstream improvements

### PR DESCRIPTION
**Pull request by gberaudo**
A remake of #38.

### CODE REVIEW

**Comment by gberaudo** [comment]
Hi @arnaud-morvan and @asaunier, I have fixed Travis. Could you now please test this branch on your side?

**Comment by gberaudo** [comment]
@asaunier, the publish should now be fixed. I see the new tag in https://hub.docker.com/r/c2corg/v6_images/tags/. Please try it and let me know if I can merge.

**Comment by gberaudo** [comment]
@asaunier, it may be due to the command you are using. Could you try modifying the docker-compose.yml this way:
- remove the `command` line;
- add to environments a new line: `GUNICORN_PARAMS: '-w 4 -b :8080 --capture-output --log-config logging.conf --log-syslog --log-syslog-to udp://syslog:514 --log-level debug -k gaiohttp'`

**Comment by asaunier** [comment]
@gberaudo Thanks for the instructions!
It seems a .swp file is in the way when I try to open the docker-compose.yml file on the demo. Do you have it opened on your side?

I have edited the file anyway, making the changes you requested. The container is now Up :)
But I still get a 503 error when trying to upload a file with the new docker image.

**Comment by gberaudo** [comment]
@asaunier, yes closed it.
There is an issue with logging.conf:
```
Error: Error: log config 'logging.conf' not found
```
Where is this logging.conf file supposed to be? I could not find it.

I removed the parameter and the aiohttp kernel and the container looks to start correctly.

What is the URL of the demo server? http://www.demov6.camptocamp.org/ ?
And the password?

**Comment by asaunier** [comment]
> Where is this logging.conf file supposed to be? I could not find it.

No clue.

> What is the URL of the demo server? http://www.demov6.camptocamp.org/ ?

Right, username/password is c2corg/c2corg

**Comment by asaunier** [comment]
You can try adding images to this (collaborative) document: http://www.demov6.camptocamp.org/routes/54993/fr/mont-blanc-du-tacul-supercouloir

**Comment by gberaudo** [comment]
Latest error: 'region' looks like the S3 region.
There was no configuration in the docker-compose.yml, I added:
```
      AWS_DEFAULT_REGION: eu-west-1
      S3_SIGNATURE_VERSION: s3v4
```

You should try with and without the signature line.

Recaptcha thinks I am a robot and other issues, @asaunier could you please try again sending an image?

**Comment by gberaudo** [comment]
@mfournier,
- do you know about this logging.conf file?
- the aiohttp backend was disabled, do we need it? It seems to require the aiohttp-wsgi library which has no version for python 3.6 (https://pypi.python.org/pypi/aiohttp-wsgi/0.7.0).

**Comment by gberaudo** [comment]
@asaunier, I digged more and discussed with @mfournier:
- the logging.conf was restored;
- the aiohttp technique was dropped (it was experimental and [wsgi support for was removed by the aiohttp project](http://aiohttp.readthedocs.io/en/stable/migration.html#wsgi)).

So the PR is now ready for merging.

**Comment by asaunier** [comment]
@gberaudo It seems the photo backend has already been upgraded in c2corg demo, right?

@fjacon @desnoes I have made a few tests and it seems to work on the demo. I had a quick look at the code changes and see no problem but I don't know much about this application and would prefer that @arnaud-morvan makes the review.

**Comment by gberaudo** [comment]
Yes @asaunier, I upgraded it yesterday so that you can test.

File: c2corg_images/__init__.py [hunk 1]
```diff
@@ -25,16 +33,23 @@ def cors_headers(request, response):
...
-config.add_route('publish', '/publish')
-config.add_route('delete', '/delete')
-config.add_static_view(name='active', path=os.environ['ACTIVE_FOLDER'])
-config.add_subscriber(add_cors_headers_response_callback, NewRequest)
+def main(_, **settings):
+    """ This function returns a Pyramid WSGI application.
+    """
+    config = Configurator(settings=settings, route_prefix=os.environ.get('ROUTE_PREFIX', '/'))
+    config.include(c2cwsgiutils.pyramid.includeme)
+    config.add_renderer('myjson', JSON())
```
Comment (arnaud-morvan) [inline c2corg_images/__init__.py]: seems that this could be removed

**Comment by arnaud-morvan** [comment]
I do not followed precisely this subject, but this code seems really good, problably already reviewed on suisse alpine side. If it has been successfully tested on c2corgv6 demo, is seems everythings go fine.
Ok for me to merge this.

File: c2corg_images/__init__.py (hunk 1 above)
Comment (gberaudo) [reply c2corg_images/__init__.py]: Indeed, removing it.

**Comment by gberaudo** [comment]
Thanks for the review @arnaud-morvan. I have addressed your comment about the unused myjson render and will merge when Travis is green.

**Comment by asaunier** [comment]
@gberaudo @arnaud-morvan  I would like to make a new release for ``v6_images``. See https://github.com/c2corg/v6_images/releases

OK for you? What version number do you recommend? v6.0.5? v6.1.0?

**Comment by arnaud-morvan** [comment]
+1 for v6.1.0

**Comment by asaunier** [comment]
OK I have released version v6.1.0.

By the way I have noticed that the image is much bigger than previously:
* v6.1.0: 472 MB
* v6.0.4: 149 MB
See https://hub.docker.com/r/c2corg/v6_images/tags/

Any clue why this change?

**Comment by gberaudo** [comment]
> would like to make a new release for v6_images. See 
@asaunier, sure please do.

> By the way I have noticed that the image is much bigger than previously
Yes, the image is now based on the c2cwsgiutil image.

**Comment by asaunier** [comment]
And the c2cwsgiutil image is 300MB bigger?

**Comment by gberaudo** [comment]
It installs several dev packages, see https://github.com/camptocamp/c2cwsgiutils/blob/master/Dockerfile.
@pvalsecc, why do we need these?

**Comment by asaunier** [comment]
I also notice that the image (container?) sizes varies a lot on our demo server:
```
root@demo:~/docker-stuff/composition/demov6# docker images
REPOSITORY              TAG                 IMAGE ID            CREATED             SIZE
c2corg/v6_images        v6.1.0              8b771be0ed92        About an hour ago   1.339 GB
c2corg/v6_images        to_upstream         0bef1dc4eb98        6 days ago          921.4 MB
...
```

**Comment by gberaudo** [comment]
@asaunier, maybe related to this recent change in c2cwsgiutils: https://github.com/camptocamp/c2cwsgiutils/pull/16.

**Comment by asaunier** [comment]
OK, I have a bit the feeling that having those images getting such fatter might become a problem, don't you think?

**Comment by gberaudo** [comment]
I expect it should have low impact (only disk space).

On the bright side, it brings new features, that could be useful to the ui image too. If you base it on c2cwsgiutils, it will become a common base layer, shared between the images.

**Comment by arnaud-morvan** [comment]
yes, regarding the docker history of c2cwsgiutils image, it is the gdal layer that is the biggest.

**Comment by pvalsecc** [comment]
The image size should be no problem. We are talking of 100s of megs of disk space, here...

**Comment by pvalsecc** [comment]
@arnaud-morvan: Good news for you, I've found a few problems in the Dockerfile of python-gis and greatly reduced its size: https://github.com/camptocamp/docker-python-gis/pull/1

**Pull request merged**